### DISCUSSION TITLE
Selective supplemental groups

### DISCUSSION THREAD

**Discussion by netstickbat** [containers/podman#19016] [General]
Is there any way of being more selective of what supplemental host groups can be shared with a container? Currently, it seems that all of them are shared with `--group-add keep-groups` and I reckon that does increase a container's attack surface in the case of an escape.

**Comment by rhatdan** [comment]
No, we can either leak all groups into the container or only the groups defined in /etc/subgid for the user.
This is controlled by the kernel.

**Comment by netstickbat** [comment]
Thanks for the reply, @rhatdan! However I am not sure if I follow after rereading both subgid(5) and newgidmap(1). Just to make sure I am getting my point across correctly; basically I have a user on the host that belongs to the following groups:

```
$ groups
alpine wheel audio video usb
```

What I am after is to use a USB device as controlled by the `usb` group on the host inside a container so I'd need, for instance, `--device /dev/hidraw0`. Still, that doesn't understandably allow one to use it there as I don't have group access, so I'd then need `--group-add keep-groups` as explained [here](https://github.com/containers/podman/blob/main/troubleshooting.md#20-passed-in-devices-or-files-cant-be-accessed-in-rootless-container).

As you mentioned, that leaks all groups into the container, so in case a container process manages to escape, it would then have the necessary permissions to access the resources granted by the remaining groups.

Having said that, I have tried to fiddle with `/etc/subgid` to no avail. Do you mind shedding some light on how it could be used to accomplish this, if it's actually possible at all?

**Reply by Luap99** [reply]
This issue might help you: https://github.com/containers/podman/issues/18333#issuecomment-1608880995
